src/toxcore/dht/kbucket.rs

```diff
@@ -103,38 +97,18 @@ pub struct Bucket {
 }
 
 /// Default number of nodes that bucket can hold.
-pub const BUCKET_DEFAULT_SIZE: usize = 8;
+pub const BUCKET_DEFAULT_SIZE: u8 = 8;
 
 impl Bucket {
     /** Create a new `Bucket` to store nodes close to the `PublicKey`.
 
-    Can hold up to `num` nodes if number is supplied. If `None` is
-    supplied, holds up to [`BUCKET_DEFAULT_SIZE`] nodes. If `Some(0)`
-    is supplied, it is treated as `None`.
-
-    [`BUCKET_DEFAULT_SIZE`]: ./constant.BUCKET_DEFAULT_SIZE.html
+    Can hold up to `capacity` nodes.
     */
-    pub fn new(num: Option<u8>) -> Self {
-        trace!(target: "Bucket", "Creating a new Bucket.");
-        match num {
-            None => {
-                trace!("Creating a new Bucket with default capacity.");
-                Bucket {
-                    capacity: BUCKET_DEFAULT_SIZE as u8,
-                    nodes: Vec::with_capacity(BUCKET_DEFAULT_SIZE),
-                }
-            },
-            Some(0) => {
-                debug!("Treating Some(0) as None");
-                Bucket::new(None)
-            },
-            Some(n) => {
-                trace!("Creating a new Bucket with capacity: {}", n);
-                Bucket {
-                    capacity: n,
-                    nodes: Vec::with_capacity(n as usize),
-                }
-            }
+    pub fn new(capacity: u8) -> Self {
+        trace!("Creating a new Bucket with capacity: {}", capacity);
+        Bucket {
+            capacity,
```

Comment:
Do we need this field `capacity`?

Reply:
Regarding clone(), I agree on keeping this member variable. Bucket and Kbucket could omit deriving Clone, but it may be a problem in the future.